### DISCUSSION TITLE
"Uknown error occurred" when trying to add entity during setup..

### DISCUSSION THREAD

**Discussion by incline02** [sca075/mqtt_vacuum_camera#276] [Q&A] [answered]
### Checklist

- [X] I have updated the integration to the latest version available
- [X] I have checked if the problem is already reported

### The problem

I have installed the latest firmware to everything including HA, the dreame L40, the integration, etc.

I followed the instruction to install it, restarted HA. My dreame integration works because I can use it jointly in my homekit integration. I also see the robot on HA dashboard so I know they work.

Now I want to expose the bot maps as cameras using MQTT Vacuum Camera. When going to HA>System>Devices Integration and click + Add integration, MQTT is listed there and I click it. I then enter the domain.entity to the dreame robot : vacuum.bot. The error returned is "unknown error occurred"

### What version of an integration has described problem?

v2024.11.0

### What vacuum model do you have problems with?

Dreame L40

### Please firmware installed on your Vacuum.

4.3.9_1536

### What type of platform you use?

ARM (Raspberry Pi, Odroid, etc.) > 4GB

### What version of Home Assistant do you use?

core- 2024.11.2

### What type of installation are you running?

Home Assistant OS

### Function, that in your opinion is creating the issue.

None

**Comment by sca075** [comment]
@incline02 I would need more data to understand better what you are reporting.

Vacuum.bot is the vacuum entity id?

What is the configuration of your MQTT? I mean can you please share a screenshot of the MQTT configuration of your vacuum from the valetudo interface please?

**Comment by sca075** [comment]
@incline02 I found probably the problem. 
First of all for clarity.. if the vacuum is setup correctly as you wrote.. the integration should add a camera in home assistant without errors. If there's a validation error on the created integration entry then HA will raise a Configuration Error.

Unknown error actually (and I'm try to understand why..) can happen when there's no maps data in the vacuum, and this actually is a quite rare scenario (I suppose in your case there are data for the maps).

Also in this case.. there should be a camera entry in HA. Can you please confirm?

**Comment by sca075** [comment]
This is going to move to discussion as per there are not enough data to work on it.

**Comment by sca075** [comment, marked as answer]
The user is using the Dreame integration and this is why the MQTT Vacuum Camera isn't able to render the vacuum maps, the Unknown Error will be translated as temporary as Unsupported Vacuum. Will although investigate whenever is possible to render the vacuum maps from the Dreame (whenever the data are available via MQTT).
